### PR TITLE
Mainnet cranny

### PR DESCRIPTION
**Pull request by mosonyi**
Please review especially the authorities

### CODE REVIEW

**Comment by brenzi** [comment]
If you produce the "fresh" chainspecs and commit them to the repo, you need to export them with `--raw`. Otherwise they're not deterministic. I'll fix that with a separate commit

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -124,16 +136,219 @@ pub fn local_testnet_config() -> Result<ChainSpec, String> {
...
+	token_specs: &str,
+) -> Result<ChainSpec, String> {
+	let (root, endowed, authorities) = match genesis_keys {
+		GenesisKeys::Integritee =>
+			(IntegriteeKeys::root(), vec![IntegriteeKeys::root()], IntegriteeKeys::authorities()),
+		GenesisKeys::Cranny =>
+			(CrannyKeys::root(), vec![CrannyKeys::root()], CrannyKeys::authorities()),
+	};
+
+	// Todo: Chris check wasm binary
```
Comment (brenzi) [inline node/src/chain_spec.rs]: is this question answered?

File: node/src/chain_spec.rs [hunk 2]
```diff
@@ -124,16 +136,219 @@ pub fn local_testnet_config() -> Result<ChainSpec, String> {
...
+		move || {
+			testnet_genesis(wasm_binary, authorities.clone(), root.clone(), endowed.clone(), false)
+		},
+		token_specs,
+	))
+}
+
 /// Configure initial storage state for FRAME modules.
+///
+/// Todo: rename to genesis_config
```
Comment (brenzi) [inline node/src/chain_spec.rs]: why leave a todo here?

**Review by brenzi — CHANGES_REQUESTED** [review]
basically LGTM
But please check comments and fix those open points

File: node/src/chain_spec.rs (hunk 1 above)
Comment (mosonyi) [reply node/src/chain_spec.rs]: Not yet, needs @clangenb to answer.

File: node/src/chain_spec.rs (hunk 2 above)
Comment (mosonyi) [reply node/src/chain_spec.rs]: It was a suggestion from @clangenb , we need to discuss

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -124,16 +136,219 @@ pub fn local_testnet_config() -> Result<ChainSpec, String> {
...
+		ChainType::Live,
+		testnet_constructor,
+		Vec::new(),
+		// telemetry endpoints
+		None,
+		// protocol id
+		Some("teer"),
+		// properties
+		Some(
+			// make configarble
```
Comment (clangenb) [inline node/src/chain_spec.rs]: this comment we can remove now.

File: node/src/chain_spec.rs (hunk 1 above)
Comment (clangenb) [reply node/src/chain_spec.rs]: I wondered why they call it 'dev'-`WASM_BINARY`. I wanted to make sure that we don't include something that is not meant for production. But it seems fine. So we can remove the comment.

File: node/src/chain_spec.rs (hunk 2 above)
Comment (clangenb) [reply node/src/chain_spec.rs]: Yes, I think this is not only `testnet` anymore. So it is confusing to call it like that. Generic `genesis_config` is the best name IMO.

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -142,14 +357,19 @@ fn testnet_genesis(
...
+				.map(|k| {
+					if k == treasury_account_id() {
+						(k, treasury_funding)
+					} else if k == CrannyKeys::root() {
+						(k, 10_000_000__000_000_000_000)
+					} else {
+						(k, ENDOWED_FUNDING)
+					}
+				})
+				.collect(),
```
Comment (clangenb) [inline node/src/chain_spec.rs:368]: Congrats to your first homegrown rust-snippet! 🥳

File: node/src/chain_spec.rs [hunk 5]
```diff
@@ -19,6 +23,14 @@ pub fn get_from_seed<TPublic: Public>(seed: &str) -> <TPublic::Pair as Pair>::Pu
 		.public()
 }
 
+pub fn public_from_ss58<TPublic: Public + FromStr>(ss58: &str) -> TPublic
+where
+	// what's up with this weird trait bound??
```
Comment (clangenb) [inline node/src/chain_spec.rs]: This trait-bound comment we can also remove. It is correct that we need to put it there.

File: node/src/chain_spec.rs (hunk 4 above)
Comment (mosonyi) [reply node/src/chain_spec.rs:368]: Thanks :)

File: node/src/chain_spec.rs (hunk 5 above)
Comment (mosonyi) [reply node/src/chain_spec.rs]: comment removed

File: node/src/chain_spec.rs (hunk 3 above)
Comment (mosonyi) [reply node/src/chain_spec.rs]: comment removed

**Comment by mosonyi** [comment]
Moved the PR into a cleaner branch
https://github.com/integritee-network/integritee-node/pull/52

**Pull request closed without merging**